File: app/src/main/java/com/example/newsapp/api/ArticleNewsService.java
package com.example.newsapp.api;

import com.example.newsapp.model.ResultNews;

import io.reactivex.rxjava3.core.Single;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface ArticleNewsService {
    @GET("news")
    Single<ResultNews> getAllArticleNews(
            @Query("apikey") String apikey,
            @Query("category") String category,
            @Query("country") String country);
}

File: app/src/main/java/com/example/newsapp/api/ArticleNewsService.kt
package com.example.newsapp.api

import com.example.newsapp.model.NewsResult
import retrofit2.http.GET
import com.example.newsapp.utils.Constants
import retrofit2.Response
import retrofit2.http.Query

interface ArticleNewsService {
    @GET("news")
    suspend fun getAllArticleNews(
        @Query("category") category: String = "top",
        @Query("country") country: String = "us",
        @Query("page") pageNumber: Int = 0,
        @Query("apikey") apikey: String = Constants.API_KEY
    ): Response<NewsResult>
}